lalrpop/src/lr1/codegen/parse_table.rs

```diff
@@ -809,6 +965,83 @@ impl<'ascent, 'grammar, W: Write> CodeGenerator<'ascent, 'grammar, W, TableDrive
         Ok(())
     }
 
+    fn error_recovery(&mut self, extra_test: &str) -> io::Result<()> {
+        let phantom_data_expr = self.phantom_data_expr();
+        
+        // First perform all reductions from the current state
+        rust!(self.out, "loop {{");
+        rust!(self.out, "let {}state = *{}states.last().unwrap() as usize;",
+              self.prefix,
+              self.prefix);
+
+        if DEBUG_PRINT {
+            rust!(self.out, r#"println!("Error recovery in state: {{}}", {}state);"#,
+                  self.prefix);
+        }
+
+        // Access the action with `error` as the lookahead
+        rust!(self.out, "let {}action = {}ACTION[({}state + 1) * {} - 1];",
+              self.prefix,
+              self.prefix,
+              self.prefix,
+              self.grammar.terminals.all.len());
+        rust!(self.out, "if {}action >= 0 {{", self.prefix);
+        rust!(self.out, "break;");
+        rust!(self.out, "}}");
+
+        if DEBUG_PRINT {
+            rust!(self.out, r#"println!("Error recovery reduces on action: {{}}", {}action);"#,
+                  self.prefix);
+        }
+
+        rust!(self.out,
+              "if let Some(r) = {}reduce({}{}action, None, &mut {}states, &mut {}symbols, {}) {{",
```

Comment:
I am curious why supply `None` here -- from skimming the code, this parameter is the lookahead start for the next token, which it seems like we *do* have, right?